starter_code/queries.md

```diff
@@ -4,76 +4,115 @@
 
-<!-- Your Code Goes Here -->
+query: {$and: [{number_of_employees: {$lt:1000 }},{founded_year:{$lt:2005}}]}
+sort: {number_of_employees:1}
+limit: 10
 
 ### 6. All the companies that don't include the `partners` field.
 
-<!-- Your Code Goes Here -->
+query:  { partners: { $exists: false } }
+
 
 ### 7. All the companies that have a null type of value on the `category_code` field.
 
-<!-- Your Code Goes Here -->
+query: { category_code: { $type: "null"} }
+
 
 ### 8. All the companies that have at least 100 employees but less than 1000. Retrieve only the `name` and `number of employees` fields.
 
-<!-- Your Code Goes Here -->
+query: {$and: [{number_of_employees:{$gte:100}},{number_of_employees:{$lt:1000}}]}
+projection: {name:1,number_of_employees:1}
+
 
 ### 9. Order all the companies by their IPO price in a descending order.
 
-<!-- Your Code Goes Here -->
+
+projection: {"ipo.valuation_amount":1}
+{"ipo.valuation_amount":-1}
+
 
 ### 10. Retrieve the 10 companies with more employees, order by the `number of employees`
 
-<!-- Your Code Goes Here -->
+sort: {number_of_employees:-1}
+limit: 10
 
 ### 11. All the companies founded on the second semester of the year. Limit your search to 1000 companies.
 
-<!-- Your Code Goes Here -->
+query: {founded_month: {$gte:7}}
+limit: 1000
 
 ### 12. All the companies founded before 2000 that have an acquisition amount of more than 10.000.000
 
-<!-- Your Code Goes Here -->
+query: {$and: [{founded_year: {$lt: 2000}},{"acquisition.price_amount":{$gt: 10000000}}]}
 
 ### 13. All the companies that have been acquired after 2010, order by the acquisition amount, and retrieve only their `name` and `acquisition` field.
 
-<!-- Your Code Goes Here -->
+query: {"acquisition.acquired_year": {$gt: 2010}}
+projection: {name:1,acquisition:1}
+sort: {"acquisition.price_amount":-1}
 
 ### 14. Order the companies by their `founded year`, retrieving only their `name` and `founded year`.
 
-<!-- Your Code Goes Here -->
+projection: {founded_year:1,name:1}
+sort: {founded_year:-1}
+
 
 ### 15. All the companies that have been founded on the first seven days of the month, including the seventh. Sort them by their `acquisition price` in a descending order. Limit the search to 10 documents.
 
-<!-- Your Code Goes Here -->
+query: {founded_day:{$lte:7}}
+sort: {"acquisition.price_amount":-1}
+limit: 10
 
 ### 16. All the companies on the 'web' `category` that have more than 4000 employees. Sort them by the amount of employees in ascending order.
 
-<!-- Your Code Goes Here -->
+query: 
+projection: 
+sort: 
+skip: 
+limit: 
 
 ### 17. All the companies whose acquisition amount is more than 10.000.000, and currency is 'EUR'.
 
-<!-- Your Code Goes Here -->
+query: 
+projection: 
+sort: 
+skip: 
+limit: 
 
 ### 18. All the companies that have been acquired on the first trimester of the year. Limit the search to 10 companies, and retrieve only their `name` and `acquisition` fields.
 
-<!-- Your Code Goes Here -->
+query: 
```

Comment:
{ $and:[{"acquisitions.price_amount": {$gt: 10000000}}, {"acquisitions.price_currency_code": {$eq: "EUR"} } ] }